evalml/automl/utils.py

```diff
@@ -134,5 +135,19 @@ def get_best_sampler_for_data(X, y, sampler_type, sampler_balanced_ratio):
     # if all class ratios are larger than the ratio provided, we don't need to sample
     if all(class_ratios >= sampler_balanced_ratio):
         return None
-    # we default to using the Undersampler
-    return 'Undersampler'
+    # We set a threshold to use the Undersampler in order to avoid long runtimes
+    elif len(y) >= 20000:
```

Comment:
Maybe we should define this as a constant like we do for `_LARGE_DATA_ROW_THRESHOLD`?